src/main/resources/avro/bdg.avdl

```diff
@@ -988,6 +703,97 @@ record VariantAnnotation {
 }
 
 /**
+ Allele encodings for genotypes.
+ */
+enum Allele {
+
+  /**
+   Reference allele.
+   */
+  REF,
+
+  /**
+   Alternate allele.
+   */
+  ALT,
+
+  /**
+   An unspecified other alternate allele. This may occur when the variant for
+   the genotype partially encoded by this allele is split as a multi-allelic site.
+   */
+  OTHER_ALT,
+
+  /**
+   Missing allele.
+   */
+  NO_CALL
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
+
+  /**
+   The variant for this genotype.
+   */
+  union { null, Variant } variant = null;
+
+  /**
```

Comment:
If we're refactoring `Genotype`, it'd be kinda nice to remove the `contigName`/`start`/`end` fields, since those are nested in `variant`. We flattened them a while back due to a performance regression in Parquet when running predicates.